### PR TITLE
Adding analyzer feedback for need for speed concept exercise

### PR DESCRIPTION
**Pull request by manumafe98**
# pull request

Related issue: [#2682](https://github.com/exercism/java/issues/2682)

As the exercise does not have loops as a prerequisites, and we do not have a concept that introduces to while loops I made some slight changes to the reference resolution as well

Also I got my inspiration for the [go analyzer](https://github.com/exercism/go/blob/main/exercises/concept/need-for-speed/.meta/design.md#analyzer)

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by sanderploegsma** [comment]
I'm reading through the original `.meta/design.md` and it seems to be a copy of the one from `elons-toy-car`. It stood out because it mentions that constructors are out-of-scope, while the exercise teaches the concept of a constructor. 😂

So I would suggest updating the entire file in this PR to address this.

File: exercises/concept/need-for-speed/.meta/design.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 - Inheritance.
 - Finalizers.
 - Method overloading.
+- Loops.
```
Comment (sanderploegsma) [inline exercises/concept/need-for-speed/.meta/design.md:20]: I don't think we need to list this here. The out-of-scope section is typically used to clarify which parts of the concept being taught are considered out of scope for the exercise. Loops are not really part of the constructors concept, so no need to mention it.

**Comment by manumafe98** [comment]
> I'm reading through the original `.meta/design.md` and it seems to be a copy of the one from `elons-toy-car`. It stood out because it mentions that constructors are out-of-scope, while the exercise teaches the concept of a constructor. 😂
> 
> So I would suggest updating the entire file in this PR to address this.

Oh, good catch, okey I would update the entire file then

File: exercises/concept/need-for-speed/.meta/design.md (hunk 1 above)
Comment (manumafe98) [reply exercises/concept/need-for-speed/.meta/design.md:20]: 👍

**Review by sanderploegsma — APPROVED** [review]
👍

**Pull request merged**